pkg/controller/persistentvolume/controller_base.go

```diff
@@ -139,14 +140,19 @@ func (ctrl *PersistentVolumeController) initializeCaches(volumeSource, claimSour
 		glog.Errorf("PersistentVolumeController can't initialize caches: %v", err)
 		return
 	}
-	volumeList, ok := volumeListObj.(*api.List)
-	if !ok {
-		glog.Errorf("PersistentVolumeController can't initialize caches, expected list of volumes, got: %+v", volumeListObj)
-		return
-	}
-	for _, volume := range volumeList.Items {
+	// reflect is used here as api.List is used by the tests and api.PersistentVolumeList
+	// is passed in at startup.
```

Comment:
This is extremely concerning. We need to change the tests to return the same thing the API returns.